flink-connector/flink/src/main/java/org/apache/gravitino/flink/connector/catalog/BaseCatalog.java

```diff
@@ -284,8 +285,11 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     String comment = table.getComment();
     Map<String, String> properties =
         propertiesConverter.toGravitinoTableProperties(table.getOptions());
+
     Transform[] partitions =
-        partitionConverter.toGravitinoPartitions(((CatalogTable) table).getPartitionKeys());
+        partitionConverter != null
```

Comment:
it's a little odd to make partitionConverter to null, how about create an specific partitionConverter for JDBC, throw exception explicitly for partition keys.